### PR TITLE
gui-apps/waybar: add 0.10.4, drop 0.9.22

### PR DESCRIPTION
**Pull request by fkobi**
just a simple bump

---

Please check all the boxes that apply:

- [X] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [X] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [X] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [X] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

**Review by MrRoy — APPROVED** [review]
LGTM, thanks !

File: gui-apps/waybar/waybar-0.10.4.ebuild [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	upower? ( logind )
+"
+
+RESTRICT="!test? ( test )"
+
+BDEPEND="
+	>=app-text/scdoc-1.9.2
+	dev-util/gdbus-codegen
+	virtual/pkgconfig
+"
```
Comment (juippis) [inline gui-apps/waybar/waybar-0.10.4.ebuild]: You need **dev-util/wayland-scanner** here as well.

File: gui-apps/waybar/waybar-0.10.4.ebuild [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+RDEPEND="
+	dev-cpp/cairomm:0
+	dev-cpp/glibmm:2
+	dev-cpp/gtkmm:3.0
+	dev-libs/glib:2
+	dev-libs/jsoncpp:=
+	dev-libs/libinput:=
```
Comment (juippis) [inline gui-apps/waybar/waybar-0.10.4.ebuild:40]: This didn't seem to be used when `-libinput` was specified. Or at least it didn't link, if it was used. Could you figure out how it's used in the build system?

File: gui-apps/waybar/waybar-0.10.4.ebuild [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	dev-libs/libsigc++:2
+	>=dev-libs/libfmt-8.1.1:=
+	>=dev-libs/spdlog-1.10.0:=
+	dev-libs/date:=
```
Comment (juippis) [inline gui-apps/waybar/waybar-0.10.4.ebuild:44]: Neither this, remove `:=`.

File: gui-apps/waybar/waybar-0.10.4.ebuild [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+	dev-libs/wayland
+	gui-libs/gtk-layer-shell
+	gui-libs/wlroots:=
+	x11-libs/gtk+:3[wayland]
+	x11-libs/libxkbcommon
+	evdev? ( dev-libs/libevdev:= )
```
Comment (juippis) [inline gui-apps/waybar/waybar-0.10.4.ebuild]: This doesn't have a subslot, remove `:=`

File: gui-apps/waybar/waybar-0.10.4.ebuild [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+	jack? ( virtual/jack )
+	libinput? ( dev-libs/libinput:= )
+	logind? (
+		|| ( sys-apps/systemd
+			 sys-auth/elogind )
+	)
+	mpd? ( media-libs/libmpdclient )
+	mpris? ( >=media-sound/playerctl-2 )
+	network? ( dev-libs/libnl:3 )
+	pipewire? ( media-video/wireplumber:0/0.5 )
```
Comment (juippis) [inline gui-apps/waybar/waybar-0.10.4.ebuild:60]: This seems to link to pipewire, please figure if the slot `>=0.5` requirement still holds true and add a subslot binder dependency for pipewire as well.

File: gui-apps/waybar/waybar-0.10.4.ebuild [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+	pulseaudio? ( media-libs/libpulse )
+	sndio? ( media-sound/sndio:= )
+	systemd? ( sys-apps/systemd:= )
+	tray? (
+		dev-libs/libdbusmenu[gtk3]
+		dev-libs/libayatana-appindicator
+	)
+	udev? ( virtual/libudev:= )
+	upower? ( sys-power/upower )
```
Comment (juippis) [inline gui-apps/waybar/waybar-0.10.4.ebuild]: Add subslot binder here since it's linking,
`:=`.

File: gui-apps/waybar/waybar-0.10.4.ebuild [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+	mpd? ( media-libs/libmpdclient )
+	mpris? ( >=media-sound/playerctl-2 )
+	network? ( dev-libs/libnl:3 )
+	pipewire? ( media-video/wireplumber:0/0.5 )
+	pulseaudio? ( media-libs/libpulse )
+	sndio? ( media-sound/sndio:= )
+	systemd? ( sys-apps/systemd:= )
+	tray? (
+		dev-libs/libdbusmenu[gtk3]
+		dev-libs/libayatana-appindicator
```
Comment (juippis) [inline gui-apps/waybar/waybar-0.10.4.ebuild]: Sort these alphabetically.

File: gui-apps/waybar/waybar-0.10.4.ebuild (hunk 3 above)
Comment (juippis) [reply gui-apps/waybar/waybar-0.10.4.ebuild:44]: See what the build system does with date - how the lib is called. Grep the source code for it.

**Review by juippis — COMMENTED** [review]
@MrRoy could you take a look and approve if it's okay in your opinion? I'd like your take on the pipewire situation, especially.

**Review by arthurzam — APPROVED** [review]

**Pull request closed without merging**